amid/msd.py

```diff
@@ -0,0 +1,155 @@
+import contextlib
+import gzip
+import json
+import os
+import tarfile
+from pathlib import Path
+
+import nibabel as nb
+import numpy as np
+from connectome import Output, Source, Transform, meta
+from connectome.interface.nodes import Silent
+
+from .internals import checksum, licenses, register
+
+
+@register(
+    body_region=('Chest', 'Abdominal', 'Head'),
+    license=licenses.CC_BYSA_40,  # check all datasets
+    link='http://medicaldecathlon.com/',
+    modality=('CT', 'CE CT', 'MRI', 'MRI FLAIR', 'MRI T1w', 'MRI t1gd', 'MRI T2w', 'MRI T2', 'MRI ADC'),
+    raw_data_size='97.8G',
+    task='Image segmentation',
+)
+@checksum('msd')
+class MSD(Source):
+    """
+    MSD is a Medical Segmentaton Decathlon Challenge with 10 tasks.
+    Parameters
+    ----------
+    root : str, Path, optional
+        path to the folder containing the raw downloaded archives.
+        If not provided, the cache is assumed to be already populated.
+    version : str, optional
+        the data version. Only has effect if the library was installed from a cloned git repository.
+    Notes
+    -----
+    Data can be downloaded here:http://medicaldecathlon.com/
+    or here: https://msd-for-monai.s3-us-west-2.amazonaws.com/
+    or here: https://drive.google.com/drive/folders/1HqEgzS8BV2c7xYNrZdEAnrHk7osJJ--2/
+    Then, the folder with raw downloaded data should contain tar archive with data and masks
+    (`Task03_Liver.tar`).
+    """
+
+    _root: str = None
+    _task_to_name: dict = {
```

Comment:
do you really need this as a layer argument?

Reply:
is there any other way to define a field in Connectome dataset? Should we do it like this?
```python
class Dataset(Source):
    def _task_to_name():
        return dict(...)
```

Reply:
not really. I would use a global constant here